tools/Umbraco.JsonSchema/UmbracoJsonSchemaGenerator.cs

```diff
@@ -0,0 +1,34 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+using NJsonSchema.Generation;
+
+/// <inheritdoc />
+public class UmbracoJsonSchemaGenerator : JsonSchemaGenerator
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UmbracoJsonSchemaGenerator" /> class.
+    /// </summary>
+    public UmbracoJsonSchemaGenerator()
+        : base(new JsonSchemaGeneratorSettings()
+        {
+            AlwaysAllowAdditionalObjectProperties = true,
+            DefaultReferenceTypeNullHandling = ReferenceTypeNullHandling.NotNull,
```

Comment:
We might also want to flatten the inheritance hierarchy to generate cleaner/smaller schemas (I've implemented this for the JSON schema generation in our other products):

```suggestion
            DefaultReferenceTypeNullHandling = ReferenceTypeNullHandling.NotNull,
            FlattenInheritanceHierarchy = true,
```